openmdao/core/problem.py

```diff
@@ -731,7 +735,8 @@ def _check_solvers(self):
                 group_states.append((group, states))
 
                 # this group has an iterative lin solver, so all states in it are ok
-                if group.ln_solver.options['maxiter'] > 1:
+                if isinstance(group.ln_solver, DirectSolver) or \
+                   group.ln_solver.options['maxiter'] > 1:
```

Comment:
looks like you just did this same test at the top of the loop... couldn't you just use `has_iter_solver[group.pathname]`?